packages/pigeon/test/java_generator_test.dart

```diff
@@ -764,4 +764,68 @@ void main() {
     expect(code, contains('doit(Reply<List<Long>> callback)'));
     expect(code, contains('List<Long> output ='));
   });
+
+  test('host multiple args', () {
+    final Root root = Root(apis: <Api>[
+      Api(name: 'Api', location: ApiLocation.host, methods: <Method>[
+        Method(
+          name: 'add',
+          arguments: <NamedType>[
+            NamedType(
+                name: 'x',
+                type: TypeDeclaration(isNullable: false, baseName: 'int')),
+            NamedType(
+                name: 'y',
+                type: TypeDeclaration(isNullable: false, baseName: 'int')),
+          ],
+          returnType: TypeDeclaration(baseName: 'int', isNullable: false),
+          isAsynchronous: false,
+        )
+      ])
+    ], classes: <Class>[], enums: <Enum>[]);
+    final StringBuffer sink = StringBuffer();
+    const JavaOptions javaOptions = JavaOptions(className: 'Messages');
+    generateJava(javaOptions, root, sink);
+    final String code = sink.toString();
+    expect(code, contains('class Messages'));
+    expect(code, contains('Long add(Long x, Long y)'));
+    expect(
+        code, contains('ArrayList<Object> args = (ArrayList<Object>)message;'));
+    expect(code, contains('Long arg0 = (Long)args.get(0)'));
+    expect(code, contains('Long arg1 = (Long)args.get(1)'));
+    expect(code, contains('Long output = api.add(arg0, arg1)'));
+  });
+
+  test('flutter multiple args', () {
+    final Root root = Root(apis: <Api>[
+      Api(name: 'Api', location: ApiLocation.flutter, methods: <Method>[
+        Method(
+          name: 'add',
+          arguments: <NamedType>[
+            NamedType(
+                name: 'x',
+                type: TypeDeclaration(isNullable: false, baseName: 'int')),
+            NamedType(
+                name: 'y',
+                type: TypeDeclaration(isNullable: false, baseName: 'int')),
+          ],
+          returnType: TypeDeclaration(baseName: 'int', isNullable: false),
+          isAsynchronous: false,
+        )
+      ])
+    ], classes: <Class>[], enums: <Enum>[]);
+    final StringBuffer sink = StringBuffer();
+    const JavaOptions javaOptions = JavaOptions(className: 'Messages');
+    generateJava(javaOptions, root, sink);
+    final String code = sink.toString();
+    expect(code, contains('class Messages'));
+    expect(
+        code,
+        contains(
+            'public void add(Long arg0, Long arg1, Reply<Long> callback)'));
```

Comment:
Why aren't these real argument names?

Reply:
I never implemented it for java.  I can do that now, it should be a small change I think.

Reply:
Oh wait, I was thinking this was for the interface.  I already do this for the interface.

The reason I didn't allow user specified names here is because I didn't want them to collide with other names in that function.  If they named the arguments `output` or `channel` for example it would break the code.

Reply:
As mentioned in the other comment, you could prefix to avoid that problem. E.g., `arg_$argName`.

Reply:
Done.  When running into this problem in non-hygienic macros you typically use something like `gensym` to make sure you get a unique symbol for the local variable.  I made a simple `_generateSymbol` function that just prefixes them with `pigeon_`.

Reply:
I'm still seeing a bunch of places in the generated code (both ObjC and Java) where things are called arg0, arg1, etc. Can't they all be prefixed versions of the names?

Reply:
Yea, I only implemented it for `@FlutterApi()` + Java.  It looks like it still happens for `@HostApi()` + Dart and some of the files just haven't been regenerated.  I should be able to duplicate what I did here across the other languages, shouldn't be a big deal.  We'll say this is done when 'arg0' doesn't show up in the diff anywheres.

Reply:
I went back through and audited everything.  All user facing variables have user supplied names.  The leftover `arg0` arguments are in generated code that don't face the user.  You mention in another comment that those could be easier to debug if they had names so I'll make that change.